### ISSUE TITLE
Allow for sub-annual obs-to-model of ice-ocean coupling analysis

### ISSUE THREAD

**Issue by stephenprice** [MPAS-Dev/MPAS-Analysis#211]
When running ice-ocean-coupling specific analysis (currently on Xylar's branch "xylar-ocean-transects") -- plotting of ocean bottom temperatures and salinity and sub-shelf melt rate -- one can only compare at annual time resolution (trying to plot the analysis for specific months returns an error). For the bottom temperature and salinity observations, this can be fixed by processing the SOSE data at monthly time resolution and adding a monthly dir structure in the obs. data dirs. For sub-shelf melt rates, sub-annual data do not exist, so a fix could include simply sym linking monthly data in monthly sub dirs back to the annual data.

**Comment by xylar** [comment]
@stephenprice, could we split these into 2 issues so they can be addressed separately?

**Comment by xylar** [comment]
@stephenprice, I have a [branch](https://github.com/xylar/MPAS-Analysis/tree/add_sose_T_S) now that supports comparing T and S at multiple depths and over any of the seasons listed in [constants.monthDictionary](https://github.com/xylar/MPAS-Analysis/blob/add_sose_T_S/mpas_analysis/shared/constants/constants.py#L25)

If you want to use this branch, you'll want to add something like the following to your custom config file:
```
[output]
generate = ['all_SOSE']

[climatologyMapSoseTemperature]
depths = [0, -200, -400, -600, -800, 'bot']
comparisonTimes =  ['JFM', 'JAS', 'ANN']

[climatologyMapSoseSalinity]
depths = [0, -200, -400, -600, -800, 'bot']
comparisonTimes =  ['JFM', 'JAS', 'ANN']
```

Let me know if the usage isn't obvious here.  Depths are in meters, or `'bot'` for the sea floor; seasons are individual months or named seasons as listed in `monthDictonary`.  Setting `generate` to `all_SOSE` above will only generate the SOSE data, so modify that as you need.

**Comment by xylar** [comment]
A couple of example plots:
Temperature at z=-600 m from winter from the first year of Jeremy's B case with ice shelves on LANL IC:
![antarctic_temp_depth_-600_sose_b_build_double_counting_jas_years0001-0001](https://user-images.githubusercontent.com/4179064/27392389-d8c050bc-56a6-11e7-8432-caf7d50d7bb9.png)

Same but salinity at z=-200 m.
![antarctic_salin_depth_-200_sose_b_build_double_counting_jas_years0001-0001](https://user-images.githubusercontent.com/4179064/27392472-0f7ed722-56a7-11e7-9134-d2f2f0c6a25e.png)

**Comment by xylar** [comment]
I'll upload the files needed to run this branch to Edison this evening (I'm still traveling back from FRISP) and update the Edison config files accordingly.

**Comment by xylar** [comment]
Wow, I just noticed that I never pushed these needed files to my remote branch.  I've just done this.  This analysis should now be available.  I'll add a description on how to get the branch and how to use it on the ALCC page.

**Comment by xylar** [comment]
Addressed by #288
